### PR TITLE
🌱 Only run allowed checks in different modes

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Replace file parsing logic for recognizing supported/unsupported checks. 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

File: docs/checks/internal/validate/main.go
Comment (laurentsimon) [inline docs/checks/internal/validate/main.go:37]: I think it's dangerous to remove some sanity checks. This code already caught one problem when someone updated a check and it would have changed the supported checks. I'm pretty sure we will update some checks; forget to update the `registerCheck()` and it will break the `--local` feature, and later the actions running on pull requests (not quite there yet though :-)) Today we don't have unit tests for the `--repo` option, which will make it hard to catch.

What can we do to add sanity checks?

Comment (azeemshaikh38) [reply docs/checks/internal/validate/main.go:37]: #435 would add e2e tests for `--local` option?

Comment (laurentsimon) [reply docs/checks/internal/validate/main.go:37]: in the pipeline, yes.

Comment (azeemshaikh38) [reply docs/checks/internal/validate/main.go:37]: We can consider prioritizing it if that addresses your concern here. I need this to unblock myself for `commit-based` Scorecard work. Would prefer getting this PR in and in parallel working to get the e2e tests for this.

Comment (laurentsimon) [reply docs/checks/internal/validate/main.go:37]: I think this should work. Let's try to avoid releasing a scorecard action before these checks are back in.

File: checks/binary_artifact.go
Comment (laurentsimon) [inline checks/binary_artifact.go:32]: fyi, when we added the `--local` repo, we decided to have the metadata in the checks.yaml rather than hardcoded in the implementation. What's the motivation for changing it?

**Review by laurentsimon — APPROVED** [review]
LGTm overall. Added a question why the type is moved from checks.yaml to the check's implementation

File: checks/binary_artifact.go
Comment (azeemshaikh38) [reply checks/binary_artifact.go:32]: > we decided to have the metadata in the checks.yaml rather than hardcoded in the implementation.

True. After our current implementation I have found it confusing that the `docs` package validates the correctness of `checks`. My main motivation is changing this is so that each check is complete in itself and can self-validate, if that makes sense. 

Let me know. Will wait on your response to submit this PR.

File: docs/checks/internal/validate/main.go
Comment (azeemshaikh38) [reply docs/checks/internal/validate/main.go:37]: SGTM.

File: checks/binary_artifact.go
Comment (oliverchang) [reply checks/binary_artifact.go:32]: Moving here seems OK to me. Assuming I'm understanding this correctly, this level of granularity like an implementation detail that users don't really have to understand. 

`local` on the other hand would be useful for users of scorecards to know and be documented in `checks.yaml`/`checks.md`.

Comment (oliverchang) [reply checks/binary_artifact.go:32]: Or actually maybe I've misunderstood this -- the plan is to move the source of truth for "local" completely to being in the implementation. That seems fine to me, and while our output `checks.md` doesn't show this today, in the future we might want to show this there as well. 

Either way, this seems fine to me.

Comment (azeemshaikh38) [reply checks/binary_artifact.go:32]: Thanks! I'm merging this in that case.

**Pull request merged**
